<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.36">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>IMPACT Example Script</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="impact-example-script_files/libs/clipboard/clipboard.min.js"></script>
<script src="impact-example-script_files/libs/quarto-html/quarto.js"></script>
<script src="impact-example-script_files/libs/quarto-html/popper.min.js"></script>
<script src="impact-example-script_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="impact-example-script_files/libs/quarto-html/anchor.min.js"></script>
<link href="impact-example-script_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="impact-example-script_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="impact-example-script_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="impact-example-script_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="impact-example-script_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#pulling-data" id="toc-pulling-data" class="nav-link active" data-scroll-target="#pulling-data">Pulling Data</a></li>
  <li><a href="#processing-data" id="toc-processing-data" class="nav-link" data-scroll-target="#processing-data">Processing Data</a>
  <ul class="collapse">
  <li><a href="#check-mutation-status" id="toc-check-mutation-status" class="nav-link" data-scroll-target="#check-mutation-status">Check Mutation Status</a></li>
  <li><a href="#zero-alteration-patients" id="toc-zero-alteration-patients" class="nav-link" data-scroll-target="#zero-alteration-patients">Zero Alteration Patients</a></li>
  <li><a href="#create-gene-binary-matrix" id="toc-create-gene-binary-matrix" class="nav-link" data-scroll-target="#create-gene-binary-matrix">Create Gene Binary Matrix</a></li>
  <li><a href="#confirm-zero-patients-are-included-in-binary-matrix" id="toc-confirm-zero-patients-are-included-in-binary-matrix" class="nav-link" data-scroll-target="#confirm-zero-patients-are-included-in-binary-matrix">Confirm Zero Patients Are Included in Binary Matrix</a></li>
  <li><a href="#annotate-nas-in-impact-panels" id="toc-annotate-nas-in-impact-panels" class="nav-link" data-scroll-target="#annotate-nas-in-impact-panels">Annotate NAs in IMPACT Panels</a>
  <ul class="collapse">
  <li><a href="#recode-aliases" id="toc-recode-aliases" class="nav-link" data-scroll-target="#recode-aliases">Recode Aliases</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#analyzing-data" id="toc-analyzing-data" class="nav-link" data-scroll-target="#analyzing-data">Analyzing Data</a></li>
  <li><a href="#visualizing-data" id="toc-visualizing-data" class="nav-link" data-scroll-target="#visualizing-data">Visualizing Data</a>
  <ul class="collapse">
  <li><a href="#get-tmb" id="toc-get-tmb" class="nav-link" data-scroll-target="#get-tmb">Get TMB</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">IMPACT Example Script</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="pulling-data" class="level1">
<h1>Pulling Data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cbioportalR)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.3.6      ✔ purrr   0.3.4 
✔ tibble  3.1.8      ✔ dplyr   1.0.10
✔ tidyr   1.2.0      ✔ stringr 1.4.1 
✔ readr   2.1.2      ✔ forcats 0.5.2 
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#devtools::install_github("MSKCC-Epi-Bio/gnomeR")</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gnomeR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   ggplot2

Attaching package: 'gnomeR'

The following object is masked from 'package:cbioportalR':

    impact_gene_info</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gtsummary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set_cbioportal_db</span>(<span class="st">"public"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ You are successfully connected!</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ base_url for this R session is now set to "www.cbioportal.org/api" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">test_cbioportal_db</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>✔ You are successfully connected!</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>all_studies <span class="ot">&lt;-</span> <span class="fu">available_studies</span>()</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>all_studies <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 13
  studyId    name  descr…¹ publi…² groups status impor…³ allSa…⁴ readP…⁵ cance…⁶
  &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;     &lt;int&gt; &lt;lgl&gt;   &lt;chr&gt;  
1 acc_tcga   Adre… "TCGA … TRUE    PUBLIC      0 2022-0…      92 TRUE    acc    
2 bcc_unige… Basa… "Whole… TRUE    PUBLIC      0 2022-0…     293 TRUE    bcc    
3 ampca_bcm… Ampu… "Exome… TRUE    PUBLIC      0 2022-0…     160 TRUE    ampca  
4 blca_dfar… Blad… "Whole… TRUE    PUBLIC      0 2022-0…      50 TRUE    blca   
5 blca_mskc… Blad… "Compr… TRUE    PUBLIC      0 2022-0…      97 TRUE    blca   
6 blca_bgi   Blad… "Whole… TRUE    PUBLIC      0 2022-0…      99 TRUE    blca   
# … with 3 more variables: referenceGenome &lt;chr&gt;, pmid &lt;chr&gt;, citation &lt;chr&gt;,
#   and abbreviated variable names ¹​description, ²​publicStudy, ³​importDate,
#   ⁴​allSampleCount, ⁵​readPermission, ⁶​cancerTypeId</code></pre>
</div>
</div>
<p>We will use the following two studies for examples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>all_studies <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(studyId <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"blca_nmibc_2017"</span>, <span class="st">"prad_msk_2019"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 13
  studyId    name  descr…¹ publi…² groups status impor…³ allSa…⁴ readP…⁵ cance…⁶
  &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;     &lt;int&gt; &lt;lgl&gt;   &lt;chr&gt;  
1 blca_nmib… Nonm… IMPACT… TRUE    PUBLIC      0 2022-0…     105 TRUE    blca   
2 prad_msk_… Pros… MSK-IM… TRUE    PUBLIC      0 2022-0…      18 TRUE    prosta…
# … with 3 more variables: referenceGenome &lt;chr&gt;, pmid &lt;chr&gt;, citation &lt;chr&gt;,
#   and abbreviated variable names ¹​description, ²​publicStudy, ³​importDate,
#   ⁴​allSampleCount, ⁵​readPermission, ⁶​cancerTypeId</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_study_info</span>(<span class="st">"blca_nmibc_2017"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                            [,1]                                                                           
name                        "Nonmuscle Invasive Bladder Cancer (MSK Eur Urol 2017)"                        
description                 "IMPACT sequencing of 105 High Risk Nonmuscle Invasive Bladder Cancer samples."
publicStudy                 "TRUE"                                                                         
pmid                        "28583311"                                                                     
citation                    "Pietzak et al. Eur Urol 2017"                                                 
groups                      "PUBLIC"                                                                       
status                      "0"                                                                            
importDate                  "2022-08-17 22:59:50"                                                          
allSampleCount              "105"                                                                          
sequencedSampleCount        "105"                                                                          
cnaSampleCount              "105"                                                                          
mrnaRnaSeqSampleCount       "0"                                                                            
mrnaRnaSeqV2SampleCount     "0"                                                                            
mrnaMicroarraySampleCount   "0"                                                                            
miRnaSampleCount            "0"                                                                            
methylationHm27SampleCount  "0"                                                                            
rppaSampleCount             "0"                                                                            
massSpectrometrySampleCount "0"                                                                            
completeSampleCount         "0"                                                                            
readPermission              "TRUE"                                                                         
studyId                     "blca_nmibc_2017"                                                              
cancerTypeId                "blca"                                                                         
cancerType.name             "Bladder Urothelial Carcinoma"                                                 
cancerType.dedicatedColor   "Yellow"                                                                       
cancerType.shortName        "BLCA"                                                                         
cancerType.parent           "bladder"                                                                      
cancerType.cancerTypeId     "blca"                                                                         
referenceGenome             "hg19"                                                                         </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>s1 <span class="ot">&lt;-</span> <span class="fu">available_samples</span>(<span class="st">"blca_nmibc_2017"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sampleId, patientId, studyId) <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>s2 <span class="ot">&lt;-</span> <span class="fu">available_samples</span>(<span class="st">"prad_msk_2019"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sampleId,  patientId, studyId) <span class="sc">%&gt;%</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>df_pairs <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(s1, s2) <span class="sc">%&gt;%</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>patientId)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>df_pairs <span class="ot">&lt;-</span> df_pairs <span class="sc">%&gt;%</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"sample_id"</span> <span class="ot">=</span> sampleId,</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">"study_id"</span> <span class="ot">=</span> studyId)</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>all_genomic <span class="ot">&lt;-</span> <span class="fu">get_genetics_by_sample</span>(<span class="at">sample_study_pairs =</span> df_pairs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "study_id"
The following parameters were used in query:
Study ID: "blca_nmibc_2017" and "prad_msk_2019"
Molecular Profile ID: blca_nmibc_2017_mutations and prad_msk_2019_mutations
Genes: "All available genes"
Joining, by = "study_id"
The following parameters were used in query:
Study ID: "blca_nmibc_2017" and "prad_msk_2019"
Molecular Profile ID: blca_nmibc_2017_cna and prad_msk_2019_cna
Genes: "All available genes"
Joining, by = "study_id"
The following parameters were used in query:
Study ID: "blca_nmibc_2017" and "prad_msk_2019"
Molecular Profile ID: blca_nmibc_2017_structural_variants and
prad_msk_2019_structural_variants
Genes: "All available genes"</code></pre>
</div>
</div>
</section>
<section id="processing-data" class="level1">
<h1>Processing Data</h1>
<p>Some research samples included in this study. We may want to check or ask if these were sampled on different (non IMPACT panels) and also we should get distributions of alterations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>s1_all <span class="ot">&lt;-</span> <span class="fu">available_samples</span>(<span class="st">"blca_nmibc_2017"</span>) </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>s1_all <span class="ot">&lt;-</span> s1_all<span class="sc">$</span>sampleId</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>s1_all[<span class="dv">90</span><span class="sc">:</span><span class="dv">105</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "P-0008234-T01-IM5" "P-0008240-T01-IM5" "P-0008734-T01-IM5"
 [4] "P-0008834-T01-IM5" "P-0008867-T01-IM5" "P-0008880-T01-IM5"
 [7] "P-0009371-T01-IM5" "P-0010271-T01-IM5" "P-0010316-T01-IM5"
[10] "P-0010531-T01-IM5" "s_DS_nmibc_007_P"  "s_DS_nmibc_008_P" 
[13] "s_DS_nmibc_010_P"  "s_DS_nmibc_011_P"  "s_DS_nmibc_014_P" 
[16] "s_DS_nmibc_019_P" </code></pre>
</div>
</div>
<p>What type of genomic data is available?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">available_profiles</span>(<span class="st">"blca_nmibc_2017"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 8
  molecularAlterationType datatype name  descr…¹ showP…² patie…³ molec…⁴ studyId
  &lt;chr&gt;                   &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;   &lt;lgl&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;chr&gt;  
1 COPY_NUMBER_ALTERATION  DISCRETE Puta… Copy N… TRUE    FALSE   blca_n… blca_n…
2 MUTATION_EXTENDED       MAF      Muta… Mutati… TRUE    FALSE   blca_n… blca_n…
3 STRUCTURAL_VARIANT      SV       Stru… Struct… TRUE    FALSE   blca_n… blca_n…
# … with abbreviated variable names ¹​description, ²​showProfileInAnalysisTab,
#   ³​patientLevel, ⁴​molecularProfileId</code></pre>
</div>
</div>
<p>Get all genomic data:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>all_genomic <span class="ot">&lt;-</span> <span class="fu">get_genetics_by_study</span>(<span class="st">"blca_nmibc_2017"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ Returning all data for the "blca_nmibc_2017_mutations" molecular profile in the "blca_nmibc_2017" study</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ Returning all data for the "blca_nmibc_2017_cna" molecular profile in the "blca_nmibc_2017" study</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ Returning all data for the "blca_nmibc_2017_structural_variants" molecular profile in the "blca_nmibc_2017" study</code></pre>
</div>
</div>
<p>Note, you can also pull by Sample ID/Study ID pairs.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>s1 <span class="ot">&lt;-</span> <span class="fu">available_samples</span>(<span class="st">"blca_nmibc_2017"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sampleId, patientId, studyId) <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>s2 <span class="ot">&lt;-</span> <span class="fu">available_samples</span>(<span class="st">"prad_msk_2019"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sampleId,  patientId, studyId) <span class="sc">%&gt;%</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>df_pairs <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(s1, s2) <span class="sc">%&gt;%</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>patientId)</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>df_pairs <span class="ot">&lt;-</span> df_pairs <span class="sc">%&gt;%</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"sample_id"</span> <span class="ot">=</span> sampleId,</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">"study_id"</span> <span class="ot">=</span> studyId)</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>all_genomic_by_sample <span class="ot">&lt;-</span> <span class="fu">get_genetics_by_sample</span>(<span class="at">sample_study_pairs =</span> df_pairs, </span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">genes =</span> <span class="st">"TP53"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining, by = "study_id"
The following parameters were used in query:
Study ID: "blca_nmibc_2017" and "prad_msk_2019"
Molecular Profile ID: blca_nmibc_2017_mutations and prad_msk_2019_mutations
Genes: "TP53"
Joining, by = "study_id"
The following parameters were used in query:
Study ID: "blca_nmibc_2017" and "prad_msk_2019"
Molecular Profile ID: blca_nmibc_2017_cna and prad_msk_2019_cna
Genes: "TP53"
Joining, by = "study_id"
The following parameters were used in query:
Study ID: "blca_nmibc_2017" and "prad_msk_2019"
Molecular Profile ID: blca_nmibc_2017_structural_variants and
prad_msk_2019_structural_variants
Genes: "TP53"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>mutations <span class="ot">&lt;-</span> all_genomic<span class="sc">$</span>mutation</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>cna <span class="ot">&lt;-</span> all_genomic<span class="sc">$</span>cna</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>fusions <span class="ot">&lt;-</span> all_genomic<span class="sc">$</span>structural_variant</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Explore maf format (one row per mutation)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>mutations <span class="sc">%&gt;%</span> <span class="fu">str</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [1,595 × 33] (S3: tbl_df/tbl/data.frame)
 $ hugoGeneSymbol       : chr [1:1595] "ABL1" "ABL1" "ABL1" "ABL1" ...
 $ entrezGeneId         : int [1:1595] 25 25 25 25 25 25 25 142 142 207 ...
 $ uniqueSampleKey      : chr [1:1595] "UC0wMDA1NjkwLVQwMS1JTTU6YmxjYV9ubWliY18yMDE3" "UC0wMDAzNDMzLVQwMS1JTTU6YmxjYV9ubWliY18yMDE3" "UC0wMDAzNDMzLVQwMS1JTTU6YmxjYV9ubWliY18yMDE3" "UC0wMDAzMjYxLVQwMS1JTTU6YmxjYV9ubWliY18yMDE3" ...
 $ uniquePatientKey     : chr [1:1595] "UC0wMDA1NjkwOmJsY2Ffbm1pYmNfMjAxNw" "UC0wMDAzNDMzOmJsY2Ffbm1pYmNfMjAxNw" "UC0wMDAzNDMzOmJsY2Ffbm1pYmNfMjAxNw" "UC0wMDAzMjYxOmJsY2Ffbm1pYmNfMjAxNw" ...
 $ molecularProfileId   : chr [1:1595] "blca_nmibc_2017_mutations" "blca_nmibc_2017_mutations" "blca_nmibc_2017_mutations" "blca_nmibc_2017_mutations" ...
 $ sampleId             : chr [1:1595] "P-0005690-T01-IM5" "P-0003433-T01-IM5" "P-0003433-T01-IM5" "P-0003261-T01-IM5" ...
 $ patientId            : chr [1:1595] "P-0005690" "P-0003433" "P-0003433" "P-0003261" ...
 $ studyId              : chr [1:1595] "blca_nmibc_2017" "blca_nmibc_2017" "blca_nmibc_2017" "blca_nmibc_2017" ...
 $ center               : chr [1:1595] "MSKCC" "MSKCC" "MSKCC" "MSKCC" ...
 $ mutationStatus       : chr [1:1595] "SOMATIC" "SOMATIC" "SOMATIC" "SOMATIC" ...
 $ validationStatus     : chr [1:1595] "Unknown" "Unknown" "Unknown" "Unknown" ...
 $ tumorAltCount        : int [1:1595] 46 226 226 89 28 82 98 266 26 118 ...
 $ tumorRefCount        : int [1:1595] 529 487 447 677 159 615 742 428 348 74 ...
 $ normalAltCount       : int [1:1595] 0 0 0 1 0 0 0 0 0 0 ...
 $ normalRefCount       : int [1:1595] 413 379 356 464 198 385 892 346 284 359 ...
 $ startPosition        : int [1:1595] 133738181 133760387 133760412 133760819 133759700 133738211 133759501 226566970 226567800 105246551 ...
 $ endPosition          : int [1:1595] 133738181 133760400 133760418 133760819 133759700 133738211 133759502 226566970 226567800 105246551 ...
 $ referenceAllele      : chr [1:1595] "C" "GCAGCCTCTGCAGG" "GAAAGCC" "G" ...
 $ proteinChange        : chr [1:1595] "T194I" "A904Efs*31" "G912_P914delinsAF" "E1048K" ...
 $ mutationType         : chr [1:1595] "Missense_Mutation" "Frame_Shift_Del" "In_Frame_Del" "Missense_Mutation" ...
 $ functionalImpactScore: chr [1:1595] "" "" "" "" ...
 $ fisValue             : num [1:1595] 1.4e-45 1.4e-45 1.4e-45 1.4e-45 1.4e-45 ...
 $ linkXvar             : chr [1:1595] "" "" "" "" ...
 $ linkPdb              : chr [1:1595] "" "" "" "" ...
 $ linkMsa              : chr [1:1595] "" "" "" "" ...
 $ ncbiBuild            : chr [1:1595] "GRCh37" "GRCh37" "GRCh37" "GRCh37" ...
 $ variantType          : chr [1:1595] "SNP" "DEL" "DEL" "SNP" ...
 $ keyword              : chr [1:1595] "ABL1 T194 missense" "ABL1 truncating" NA "ABL1 E1048 missense" ...
 $ chr                  : chr [1:1595] "9" "9" "9" "9" ...
 $ variantAllele        : chr [1:1595] "T" "-" "CTTT" "A" ...
 $ refseqMrnaId         : chr [1:1595] "NM_005157.4" "NM_005157.4" "NM_005157.4" "NM_005157.4" ...
 $ proteinPosStart      : int [1:1595] 194 904 912 1048 675 204 608 540 456 17 ...
 $ proteinPosEnd        : int [1:1595] 194 904 914 1048 675 204 610 540 456 17 ...</code></pre>
</div>
</div>
<section id="check-mutation-status" class="level2">
<h2 class="anchored" data-anchor-id="check-mutation-status">Check Mutation Status</h2>
<p>Sometimes you may want to code ‘Unknown’ as ‘Somatic’ (recall that somatic mutations can occur in any of the cells of the body except the germ cells and therefore are not passed on. These are often tumor-specific acquired mutations).</p>
<p>Researchers can compare tumor mutations to a matched normal sample. Tumor-normal comparisons are crucial for identifying the somatic variants that act as driver mutations in cancer progression</p>
<p>Filter out germline (unless working on germline study)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(mutations<span class="sc">$</span>mutationStatus, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
        SOMATIC SOMATIC_VS_POOL         UNKNOWN            &lt;NA&gt; 
           1557              28              10               0 </code></pre>
</div>
</div>
</section>
<section id="zero-alteration-patients" class="level2">
<h2 class="anchored" data-anchor-id="zero-alteration-patients">Zero Alteration Patients</h2>
<p>Check if patients have zero alterations and be sure to include these patients in your final data set.</p>
<p>Example, these patients below have zero mutations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(s1_all, mutations<span class="sc">$</span>sampleId)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "P-0003677-T01-IM5" "P-0005590-T01-IM5"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if you queried, no mutations would come up</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">get_mutations_by_sample</span>(<span class="at">sample_id =</span> <span class="st">"P-0003677-T01-IM5"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>The following parameters were used in query:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Study ID: "msk_impact_2017"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Molecular Profile ID: "msk_impact_2017_mutations"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Genes: "All available genes"</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
<section id="create-gene-binary-matrix" class="level2">
<h2 class="anchored" data-anchor-id="create-gene-binary-matrix">Create Gene Binary Matrix</h2>
<p>Most genomic analyses start with a binary matrix of alterations. You can use <code>create_gene_binary()</code> following function in {gnomeR} (but be sure to check results!)</p>
<p>To use {gnomeR} functions, you need to do some data manipulation first.</p>
<p>!! NOTE: This may change in the upcoming months to be more user friendly.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>mutations_clean <span class="ot">&lt;-</span> <span class="fu">rename_columns</span>(mutations)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `x` must be a one- or two-column data frame in `deframe()`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! hugoGeneSymbol renamed Hugo_Symbol
! entrezGeneId renamed Entrez_Gene_Id
! center renamed Center
! ncbiBuild renamed NCBI_Build
! chr renamed Chromosome
! startPosition renamed Start_Position
! endPosition renamed End_Position
! mutationType renamed Variant_Classification
! variantType renamed Variant_Type
! referenceAllele renamed Reference_Allele
! sampleId renamed Tumor_Sample_Barcode
! validationStatus renamed Validation_Status
! mutationStatus renamed Mutation_Status
! proteinChange renamed HGVSp_Short
! variantAllele renamed Allele</code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>cna_clean <span class="ot">&lt;-</span>cna <span class="sc">%&gt;%</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">"hugo_symbol"</span> <span class="ot">=</span> hugoGeneSymbol) <span class="sc">%&gt;%</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reformat_cna</span>()</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>fusions_clean <span class="ot">&lt;-</span> fusions <span class="sc">%&gt;%</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Hugo_Symbol"</span> <span class="ot">=</span> site1HugoSymbol) <span class="sc">%&gt;%</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_columns</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `x` must be a one- or two-column data frame in `deframe()`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! ncbiBuild renamed NCBI_Build
! sampleId renamed Tumor_Sample_Barcode</code></pre>
</div>
</div>
</section>
<section id="confirm-zero-patients-are-included-in-binary-matrix" class="level2">
<h2 class="anchored" data-anchor-id="confirm-zero-patients-are-included-in-binary-matrix">Confirm Zero Patients Are Included in Binary Matrix</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>binmat1 <span class="ot">&lt;-</span> <span class="fu">create_gene_binary</span>(</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">samples =</span> s1_all, </span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mutation =</span> mutations_clean</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(mutations_clean<span class="sc">$</span>Tumor_Sample_Barcode) <span class="sc">%&gt;%</span> <span class="fu">length</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 103</code></pre>
</div>
</div>
<p>Samples arg should be specified!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>binmat2 <span class="ot">&lt;-</span> <span class="fu">create_gene_binary</span>(</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="co">#  samples = s1_all, </span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mutation =</span> mutations_clean</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ `samples` argument is `NULL`. We will infer your cohort inclusion and resulting data frame will include all samples with at least one alteration in mutation, fusion or cna data frames</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(binmat1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 105 320</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(binmat2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 103 320</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(<span class="fu">rownames</span>(binmat1), <span class="fu">rownames</span>(binmat2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "P-0003677-T01-IM5" "P-0005590-T01-IM5"</code></pre>
</div>
</div>
<p>If you didn’t specify samples, your denominator may be under counted!</p>
</section>
<section id="annotate-nas-in-impact-panels" class="level2">
<h2 class="anchored" data-anchor-id="annotate-nas-in-impact-panels">Annotate NAs in IMPACT Panels</h2>
<p>Recall that there are several versions of the IMPACT panels. Not all genes are tested in all panels. We must account for this by inserting NAs for samples/genes that were not tested.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str_detect</span>(s1_all, <span class="st">"IM5"</span>) <span class="sc">%&gt;%</span> <span class="fu">sum</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 97</code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str_detect</span>(s1_all, <span class="st">"IM3"</span>) <span class="sc">%&gt;%</span> <span class="fu">sum</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2</code></pre>
</div>
</div>
<p>Let’s check what results look like before we specify panels:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>binmat <span class="ot">&lt;-</span> <span class="fu">create_gene_binary</span>(</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">samples =</span> s1_all, </span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mutation =</span> mutations_clean, </span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cna =</span> cna_clean, </span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">fusion =</span> fusions_clean</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(binmat[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  ABL1 PARP1 AKT1 ALK APC BIRC3
P-0001453-T01-IM3    1     0    0   0   1     0
P-0002166-T01-IM3    0     0    0   0   0     0
P-0003238-T01-IM5    0     0    0   0   0     0
P-0003257-T01-IM5    0     0    0   0   1     0
P-0003261-T01-IM5    1     0    0   0   1     0
P-0003352-T01-IM5    0     0    0   0   0     0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>binmat <span class="ot">&lt;-</span> binmat <span class="sc">%&gt;%</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"sample_id"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">map_dbl</span>(binmat, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> sample_id       ABL1      PARP1       AKT1        ALK        APC      BIRC3 
         0          0          0          0          0          0          0 
        AR       RHOA      ZFHX3        ATM        ATR       ATRX        AXL 
         0          0          0          0          0          0          0 
     BARD1      CCND1       BCL6      PRDM1        BLM     BMPR1A      FOXL2 
         0          0          0          0          0          0          0 
     BRCA1       BRAF      BRCA2       CALR      CASP8      RUNX1      CCND2 
         0          0          0          0          0          0          0 
     CCND3       CDH1       CDK8     CDKN1A     CDKN1B     CDKN2A      CEBPA 
         0          0          0          0          0          0          0 
     CHEK1     CREBBP       CRKL      CSF3R     CTNNB1       DAXX      DNMT1 
         0          0          0          0          0          0          0 
    DNMT3A     DNMT3B       E2F3       EGFR      EIF4E      EP300      EPHA3 
         0          0          0          0          0          0          0 
     EPHA5      EPHA7      EPHB1      ERBB2      ERBB3      ERBB4      ERCC2 
         0          0          0          0          0          0          0 
     ERCC4      ERCC5        ERG       ESR1       ETV1       ETV6       EZH2 
         0          0          0          0          0          0          0 
     FANCA      FANCC       FAT1       FGF4      FGFR1      FGFR3      FGFR2 
         0          0          0          0          0          0          0 
     FGFR4         FH      FOXO1       FLT1       FLT3       FLT4       MTOR 
         0          0          0          0          0          0          0 
       FYN      GATA2      GATA3       GLI1       GNAQ       GNAS       GPS2 
         0          0          0          0          0          0          0 
    GRIN2A      GSK3B       MSH6       H1-2      H3-3A      H3-3B        HGF 
         0          0          0          0          0          0          0 
     FOXA1       HRAS     DNAJB1       IDH1      IGF1R       IGF2       IL7R 
         0          0          0          0          0          0          0 
      IL10      INHBA     INPP4A       INSR      INSRR       IRF4       IRS1 
         0          0          0          0          0          0          0 
      JAK1       JAK2       JAK3        JUN        KDR        KIT       KRAS 
         0          0          0          0          0          0          0 
      LMO1     SH2D1A      SMAD2      SMAD4        MAX       MCL1       MDM2 
         0          0          0          0          0          0          0 
      MDM4     MAP3K1       MEN1        MET       MITF       MLH1      KMT2A 
         0          0          0          0          0          0          0 
       MPL       MSH2       MST1      MST1R      MUTYH        MYC       MYCL 
         0          0          0          0          0          0          0 
      MYCN      MYD88      MYOD1        NBN        NF1        NF2     NFE2L2 
         0          0          0          0          0          0          0 
    NFKBIA     NOTCH1     NOTCH2     NOTCH3     NOTCH4       NPM1       NRAS 
         0          0          0          0          0          0          0 
     NTRK1      NTRK3       PAK1     PDGFRA     PDGFRB        PGR    PIK3C2G 
         0          0          0          0          0          0          0 
    PIK3C3     PIK3CA     PIK3CB     PIK3CG     PIK3R1     PIK3R2      PLCG2 
         0          0          0          0          0          0          0 
      PMS1       PMS2      POLD1       POLE    PPP2R1A      PPP6C    PRKAR1A 
         0          0          0          0          0          0          0 
     MAPK1      PTCH1       PTEN      PTPRD      PTPRS       RAC1      RAD21 
         0          0          0          0          0          0          0 
     RAD51     RAD51C     RAD51B     RAD51D      RAD52       RAF1       RARA 
         0          0          0          0          0          0          0 
     RASA1        RB1      KDM5A        REL        RET       RHEB       RIT1 
         0          0          0          0          0          0          0 
      ROS1    RPS6KB2       SDHA       SDHD      SRSF2    SMARCA4    SMARCB1 
         0          0          0          0          0          0          0 
   SMARCD1        SMO       SOX2       SOX9        SRC      STAT3     STAT5A 
         0          0          0          0          0          0          0 
     AURKA      STK11        SYK       TBX3     TCF7L2       TERT     TGFBR1 
         0          0          0          0          0          0          0 
    TGFBR2    TMPRSS2       TP53      TRAF2       TSC1       TSC2       TSHR 
         0          0          0          0          0          0          0 
     U2AF1      KDM6A      VEGFA        VHL        WT1       XPO1       YES1 
         0          0          0          0          0          0          0 
     CXCR4      KMT2D      NCOA3      RBM10      KDM5C     ARID1A      AXIN2 
         0          0          0          0          0          0          0 
      BAP1       H3C4       H3C3       H3C8      H3C10       H3C2       SPOP 
         0          0          0          0          0          0          0 
    RAD54L       CUL3       TP63       IRS2        EED     INPP4B      FUBP1 
         0          0          0          0          0          0          0 
     BCL10     PHOX2B       H3C7    RPS6KA4      LATS1    MAP3K13       KLF4 
         0          0          0          0          0          0          0 
    RECQL4      NCOR1      IKBKE       MDC1      NUP93      KEAP1      FGF19 
         0          0          0          0          0          0          0 
     MED12       AKT3    BCL2L11      SH2B3      RAD50      IKZF1       CTCF 
         0          0          0          0          0          0          0 
     STAG2       PLK2      MALT1      PNRC1      PTPRT      CHEK2       DIS3 
         0          0          0          0          0          0          0 
      SPEN        CIC        MGA     ICOSLG     DICER1      SF3B1       BRD4 
         0          0          0          0          0          0          0 
     SUZ12     TIMM8B      LATS2      GREM1      FOXP1       BBC3      SETD2 
         0          0          0          0          0          0          0 
   ANKRD11       SUFU      CDK12     ERRFI1       TET2       BCOR      RNF43 
         0          0          0          0          0          0          0 
      SHQ1      PBRM1      ASXL2      FBXW7       PAK5     ARID1B      RPTOR 
         0          0          0          0          0          0          0 
     KMT2C      CRLF2       NSD1       COP1      PALB2       TET1     FIP1L1 
         0          0          0          0          0          0          0 
     STK40      BRIP1   ABRAXAS1     ARID5B      TRAF7     CARD11      DOT1L 
         0          0          0          0          0          0          0 
     AMER1      ASXL1      ARID2     RICTOR      H3C13      MEF2B  TACC3.fus 
         0          0          0          0          0          0          0 
 EP300.fus     NA.fus  RAB35.fus  RUNX1.fus  FANCA.fus  FGFR3.fus   BRAF.fus 
         0          0          0          0          0          0          0 
MAP2K1.fus CDKN2A.fus   MDM2.Amp    RB1.Del   ATRX.Amp  PTPRD.Del CDKN2A.Del 
         0          0          0          0          0          0          0 
   ATM.Del CDKN2B.Del  FGF19.Amp  CCND1.Amp   FGF4.Amp   FGF3.Amp INPP4A.Amp 
         0          0          0          0          0          0          0 
  DDR2.Amp   SDHC.Amp   COP1.Amp  EPHA3.Del  EP300.Del  ERBB2.Amp  CDK12.Amp 
         0          0          0          0          0          0          0 
 FGFR1.Amp  FGFR1.Del    SRC.Amp  FGFR3.Amp   TBX3.Amp  U2AF1.Del  BARD1.Del 
         0          0          0          0          0          0          0 
 CXCR4.Del  SH2B3.Amp PTPN11.Amp  RAB35.Amp CDKN1B.Del   DIS3.Del  EPHA5.Del 
         0          0          0          0          0          0          0 
  PAK1.Amp  PPM1D.Amp   MCL1.Amp   RAF1.Amp    VHL.Amp   PAX5.Del  PPP6C.Del 
         0          0          0          0          0          0          0 
   EED.Amp    NF2.Amp NOTCH2.Amp   MYCL.Amp  PRDM1.Amp   E2F3.Amp NKX3-1.Del 
         0          0          0          0          0          0          0 
 SOCS1.Amp   CDK4.Amp SH2D1A.Del  STAG2.Del  FOXP1.Del GRIN2A.Amp  H3C13.Amp 
         0          0          0          0          0          0          0 
 H3C14.Amp RICTOR.Amp   SDHA.Amp   IL7R.Amp   TERT.Amp  PTCH1.Del   RIT1.Amp 
         0          0          0          0          0          0          0 
 CEBPA.Amp  CRLF2.Amp  CD274.Del   JAK2.Del  GATA3.Amp   FAT1.Del  RAD21.Del 
         0          0          0          0          0          0          0 
  XPO1.Amp NKX2-1.Del NFKBIA.Del  MAPK3.Amp    PGR.Del  CHEK1.Del  FOXA1.Amp 
         0          0          0          0          0          0          0 
 FOXA1.Del   YAP1.Del  KMT2A.Del   SDHD.Del  MRE11.Del  BIRC3.Del    CBL.Del 
         0          0          0          0          0          0          0 
 CCND3.Amp  CHEK2.Amp   IDH1.Amp 
         0          0          0 </code></pre>
</div>
</div>
<p>Use the <code>specify_panel</code> argument to take care of NA annotation for you!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>binmat_panel <span class="ot">&lt;-</span> <span class="fu">create_gene_binary</span>(</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">samples =</span> s1_all, </span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mutation =</span> mutations_clean, </span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cna =</span> cna_clean, </span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">fusion =</span> fusions_clean, </span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">specify_panel =</span> <span class="st">"impact"</span></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>→ Couldn't infer IMPACT panel version from these sample_ids, therefore no NA panel annotation will be done for these: s_DS_nmibc_007_P, s_DS_nmibc_008_P, s_DS_nmibc_010_P, s_DS_nmibc_011_P, s_DS_nmibc_014_P, and s_DS_nmibc_019_P</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">map_dbl</span>(binmat_panel, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.x))) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> ABL1 PARP1  AKT1   ALK   APC BIRC3 
    6     6     6     6     6     8 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>sample_panel_pair <span class="ot">&lt;-</span> binmat <span class="sc">%&gt;%</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_id)</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>sample_panel_pair <span class="ot">&lt;-</span> sample_panel_pair <span class="sc">%&gt;%</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">panel_id =</span> <span class="fu">case_when</span>(</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>      stringr<span class="sc">::</span><span class="fu">str_detect</span>(.data<span class="sc">$</span>sample_id, <span class="st">"-IM3"</span>) <span class="sc">~</span> <span class="st">"IMPACT341"</span>,</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>      stringr<span class="sc">::</span><span class="fu">str_detect</span>(.data<span class="sc">$</span>sample_id, <span class="st">"-IM5"</span>) <span class="sc">~</span> <span class="st">"IMPACT410"</span>,</span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>      stringr<span class="sc">::</span><span class="fu">str_detect</span>(.data<span class="sc">$</span>sample_id, <span class="st">"-IM6"</span>) <span class="sc">~</span> <span class="st">"IMPACT468"</span>,</span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>      stringr<span class="sc">::</span><span class="fu">str_detect</span>(.data<span class="sc">$</span>sample_id, <span class="st">"-IM7"</span>) <span class="sc">~</span> <span class="st">"IMPACT505"</span>,</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"IMPACT505"</span></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>binmat_panel <span class="ot">&lt;-</span> <span class="fu">create_gene_binary</span>(</span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">samples =</span> s1_all, </span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">mutation =</span> mutations_clean, </span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">cna =</span> cna_clean, </span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">fusion =</span> fusions_clean, </span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">specify_panel =</span> sample_panel_pair</span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a><span class="fu">map_dbl</span>(binmat_panel, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.x))) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">50</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  ABL1  PARP1   AKT1    ALK    APC  BIRC3     AR   RHOA  ZFHX3    ATM    ATR 
     0      0      0      0      0      2      0      0      2      0      0 
  ATRX    AXL  BARD1  CCND1   BCL6  PRDM1    BLM BMPR1A  FOXL2  BRCA1   BRAF 
     0      0      0      0      0      0      0      0      0      0      0 
 BRCA2   CALR  CASP8  RUNX1  CCND2  CCND3   CDH1   CDK8 CDKN1A CDKN1B CDKN2A 
     0      2      0      0      0      0      0      0      0      0      0 
 CEBPA  CHEK1 CREBBP   CRKL  CSF3R CTNNB1   DAXX  DNMT1 DNMT3A DNMT3B   E2F3 
     2      0      0      0      2      0      0      0      0      0      0 
  EGFR  EIF4E  EP300  EPHA3  EPHA5  EPHA7 
     0      2      0      0      0      2 </code></pre>
</div>
</div>
<section id="recode-aliases" class="level3">
<h3 class="anchored" data-anchor-id="recode-aliases">Recode Aliases</h3>
<p>recode alias argument helps check for any genes that have been renamed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>binmat_panel_recode <span class="ot">&lt;-</span> <span class="fu">create_gene_binary</span>(</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">samples =</span> s1_all, </span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mutation =</span> mutations_clean, </span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cna =</span> cna_clean, </span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">fusion =</span> fusions_clean, </span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">specify_panel =</span> sample_panel_pair, </span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">recode_aliases =</span> <span class="cn">TRUE</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="analyzing-data" class="level1">
<h1>Analyzing Data</h1>
<p>Now that we’ve processed the data we will summarize it. You can use <code>tbl_summary()</code> or the following experimental function which has some benefits:</p>
<ul>
<li>automatically orders genes by frequency</li>
<li>allows you to specify a prevalence threshold cutoff</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tbl_genomic</span>(binmat_panel, <span class="at">freq_cutoff =</span> .<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="yvczotapng" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#yvczotapng .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#yvczotapng .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#yvczotapng .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#yvczotapng .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#yvczotapng .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yvczotapng .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#yvczotapng .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#yvczotapng .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#yvczotapng .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#yvczotapng .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#yvczotapng .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#yvczotapng .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#yvczotapng .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#yvczotapng .gt_from_md > :first-child {
  margin-top: 0;
}

#yvczotapng .gt_from_md > :last-child {
  margin-bottom: 0;
}

#yvczotapng .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#yvczotapng .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#yvczotapng .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#yvczotapng .gt_row_group_first td {
  border-top-width: 2px;
}

#yvczotapng .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#yvczotapng .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#yvczotapng .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#yvczotapng .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yvczotapng .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#yvczotapng .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#yvczotapng .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#yvczotapng .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#yvczotapng .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#yvczotapng .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-left: 4px;
  padding-right: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#yvczotapng .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#yvczotapng .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#yvczotapng .gt_left {
  text-align: left;
}

#yvczotapng .gt_center {
  text-align: center;
}

#yvczotapng .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#yvczotapng .gt_font_normal {
  font-weight: normal;
}

#yvczotapng .gt_font_bold {
  font-weight: bold;
}

#yvczotapng .gt_font_italic {
  font-style: italic;
}

#yvczotapng .gt_super {
  font-size: 65%;
}

#yvczotapng .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 75%;
  vertical-align: 0.4em;
}

#yvczotapng .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#yvczotapng .gt_indent_1 {
  text-indent: 5px;
}

#yvczotapng .gt_indent_2 {
  text-indent: 10px;
}

#yvczotapng .gt_indent_3 {
  text-indent: 15px;
}

#yvczotapng .gt_indent_4 {
  text-indent: 20px;
}

#yvczotapng .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="gt_table">
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col"><strong>Characteristic</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>N = 105</strong><sup class="gt_footnote_marks">1</sup></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TERT</td>
<td class="gt_row gt_center">77 (73%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">KDM6A</td>
<td class="gt_row gt_center">50 (48%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FGFR3</td>
<td class="gt_row gt_center">47 (45%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ARID1A</td>
<td class="gt_row gt_center">30 (29%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">PIK3CA</td>
<td class="gt_row gt_center">29 (28%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">KMT2D</td>
<td class="gt_row gt_center">25 (24%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">STAG2</td>
<td class="gt_row gt_center">22 (21%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CDKN2A</td>
<td class="gt_row gt_center">5 (4.8%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CREBBP</td>
<td class="gt_row gt_center">22 (21%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TP53</td>
<td class="gt_row gt_center">22 (21%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ERBB2</td>
<td class="gt_row gt_center">19 (18%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ATM</td>
<td class="gt_row gt_center">16 (15%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ERCC2</td>
<td class="gt_row gt_center">18 (17%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FAT1</td>
<td class="gt_row gt_center">17 (16%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ARID1B</td>
<td class="gt_row gt_center">17 (16%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">EP300</td>
<td class="gt_row gt_center">15 (14%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FBXW7</td>
<td class="gt_row gt_center">15 (14%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">PBRM1</td>
<td class="gt_row gt_center">15 (14%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">RBM10</td>
<td class="gt_row gt_center">15 (14%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">MGA</td>
<td class="gt_row gt_center">14 (14%)</td></tr>
    <tr><td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">Unknown</td>
<td class="gt_row gt_center">2</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">BAP1</td>
<td class="gt_row gt_center">13 (12%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CDKN1A</td>
<td class="gt_row gt_center">12 (11%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ERBB3</td>
<td class="gt_row gt_center">12 (11%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">KMT2C</td>
<td class="gt_row gt_center">12 (11%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">NF1</td>
<td class="gt_row gt_center">12 (11%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">BRCA2</td>
<td class="gt_row gt_center">11 (10%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">SMARCA4</td>
<td class="gt_row gt_center">11 (10%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">SPEN</td>
<td class="gt_row gt_center">11 (10%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TBX3</td>
<td class="gt_row gt_center">8 (7.6%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TSC1</td>
<td class="gt_row gt_center">11 (10%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TERT.Amp</td>
<td class="gt_row gt_center">1 (1.0%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FGFR3.Amp</td>
<td class="gt_row gt_center">2 (1.9%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ERBB2.Amp</td>
<td class="gt_row gt_center">2 (1.9%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TBX3.Amp</td>
<td class="gt_row gt_center">3 (2.9%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">STAG2.Del</td>
<td class="gt_row gt_center">2 (1.9%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CDKN2A.Del</td>
<td class="gt_row gt_center">17 (16%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ATM.Del</td>
<td class="gt_row gt_center">2 (1.9%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FAT1.Del</td>
<td class="gt_row gt_center">1 (1.0%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">EP300.Del</td>
<td class="gt_row gt_center">1 (1.0%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CDKN2B.Del</td>
<td class="gt_row gt_center">14 (13%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FGFR3.fus</td>
<td class="gt_row gt_center">1 (1.0%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CDKN2A.fus</td>
<td class="gt_row gt_center">1 (1.0%)</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">EP300.fus</td>
<td class="gt_row gt_center">1 (1.0%)</td></tr>
  </tbody>
  
  <tfoot class="gt_footnotes">
    <tr>
      <td class="gt_footnote" colspan="2"><sup class="gt_footnote_marks">1</sup> n (%)</td>
    </tr>
  </tfoot>
</table>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>binmat_panel_var <span class="ot">&lt;-</span> binmat_panel <span class="sc">%&gt;%</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sex =</span> <span class="fu">sample</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="st">"M"</span>, <span class="st">"F"</span>),</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">size =</span> <span class="fu">nrow</span>(binmat_panel), <span class="at">replace =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tbl_genomic</span>(binmat_panel_var,</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">freq_cutoff =</span> .<span class="dv">1</span>, </span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> sex,</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">freq_cutoff_by_gene  =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_p</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_q</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>add_q: Adjusting p-values with
`stats::p.adjust(x$table_body$p.value, method = "fdr")`</code></pre>
</div>
<div class="cell-output-display">

<div id="ubvcuatjjb" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#ubvcuatjjb .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ubvcuatjjb .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ubvcuatjjb .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ubvcuatjjb .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ubvcuatjjb .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ubvcuatjjb .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ubvcuatjjb .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ubvcuatjjb .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ubvcuatjjb .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ubvcuatjjb .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ubvcuatjjb .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ubvcuatjjb .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#ubvcuatjjb .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ubvcuatjjb .gt_from_md > :first-child {
  margin-top: 0;
}

#ubvcuatjjb .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ubvcuatjjb .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ubvcuatjjb .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ubvcuatjjb .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ubvcuatjjb .gt_row_group_first td {
  border-top-width: 2px;
}

#ubvcuatjjb .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ubvcuatjjb .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ubvcuatjjb .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ubvcuatjjb .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ubvcuatjjb .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ubvcuatjjb .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ubvcuatjjb .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ubvcuatjjb .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ubvcuatjjb .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ubvcuatjjb .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-left: 4px;
  padding-right: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ubvcuatjjb .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ubvcuatjjb .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ubvcuatjjb .gt_left {
  text-align: left;
}

#ubvcuatjjb .gt_center {
  text-align: center;
}

#ubvcuatjjb .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ubvcuatjjb .gt_font_normal {
  font-weight: normal;
}

#ubvcuatjjb .gt_font_bold {
  font-weight: bold;
}

#ubvcuatjjb .gt_font_italic {
  font-style: italic;
}

#ubvcuatjjb .gt_super {
  font-size: 65%;
}

#ubvcuatjjb .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 75%;
  vertical-align: 0.4em;
}

#ubvcuatjjb .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ubvcuatjjb .gt_indent_1 {
  text-indent: 5px;
}

#ubvcuatjjb .gt_indent_2 {
  text-indent: 10px;
}

#ubvcuatjjb .gt_indent_3 {
  text-indent: 15px;
}

#ubvcuatjjb .gt_indent_4 {
  text-indent: 20px;
}

#ubvcuatjjb .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="gt_table">
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col"><strong>Characteristic</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>Overall</strong>, N = 105<sup class="gt_footnote_marks">1</sup></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>F</strong>, N = 51<sup class="gt_footnote_marks">1</sup></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>M</strong>, N = 54<sup class="gt_footnote_marks">1</sup></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>p-value</strong><sup class="gt_footnote_marks">2</sup></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col"><strong>q-value</strong><sup class="gt_footnote_marks">3</sup></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TERT</td>
<td class="gt_row gt_center">77 (73%)</td>
<td class="gt_row gt_center">39 (76%)</td>
<td class="gt_row gt_center">38 (70%)</td>
<td class="gt_row gt_center">0.5</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">KDM6A</td>
<td class="gt_row gt_center">50 (48%)</td>
<td class="gt_row gt_center">26 (51%)</td>
<td class="gt_row gt_center">24 (44%)</td>
<td class="gt_row gt_center">0.5</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FGFR3</td>
<td class="gt_row gt_center">47 (45%)</td>
<td class="gt_row gt_center">23 (45%)</td>
<td class="gt_row gt_center">24 (44%)</td>
<td class="gt_row gt_center">&gt;0.9</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ARID1A</td>
<td class="gt_row gt_center">30 (29%)</td>
<td class="gt_row gt_center">16 (31%)</td>
<td class="gt_row gt_center">14 (26%)</td>
<td class="gt_row gt_center">0.5</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">PIK3CA</td>
<td class="gt_row gt_center">29 (28%)</td>
<td class="gt_row gt_center">15 (29%)</td>
<td class="gt_row gt_center">14 (26%)</td>
<td class="gt_row gt_center">0.7</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">KMT2D</td>
<td class="gt_row gt_center">25 (24%)</td>
<td class="gt_row gt_center">11 (22%)</td>
<td class="gt_row gt_center">14 (26%)</td>
<td class="gt_row gt_center">0.6</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CREBBP</td>
<td class="gt_row gt_center">22 (21%)</td>
<td class="gt_row gt_center">8 (16%)</td>
<td class="gt_row gt_center">14 (26%)</td>
<td class="gt_row gt_center">0.2</td>
<td class="gt_row gt_center">0.7</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">STAG2</td>
<td class="gt_row gt_center">22 (21%)</td>
<td class="gt_row gt_center">10 (20%)</td>
<td class="gt_row gt_center">12 (22%)</td>
<td class="gt_row gt_center">0.7</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TP53</td>
<td class="gt_row gt_center">22 (21%)</td>
<td class="gt_row gt_center">10 (20%)</td>
<td class="gt_row gt_center">12 (22%)</td>
<td class="gt_row gt_center">0.7</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ERBB2</td>
<td class="gt_row gt_center">19 (18%)</td>
<td class="gt_row gt_center">10 (20%)</td>
<td class="gt_row gt_center">9 (17%)</td>
<td class="gt_row gt_center">0.7</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ERCC2</td>
<td class="gt_row gt_center">18 (17%)</td>
<td class="gt_row gt_center">12 (24%)</td>
<td class="gt_row gt_center">6 (11%)</td>
<td class="gt_row gt_center">0.092</td>
<td class="gt_row gt_center">0.4</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ARID1B</td>
<td class="gt_row gt_center">17 (16%)</td>
<td class="gt_row gt_center">11 (22%)</td>
<td class="gt_row gt_center">6 (11%)</td>
<td class="gt_row gt_center">0.15</td>
<td class="gt_row gt_center">0.5</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CDKN2A.Del</td>
<td class="gt_row gt_center">17 (16%)</td>
<td class="gt_row gt_center">6 (12%)</td>
<td class="gt_row gt_center">11 (20%)</td>
<td class="gt_row gt_center">0.2</td>
<td class="gt_row gt_center">0.7</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FAT1</td>
<td class="gt_row gt_center">17 (16%)</td>
<td class="gt_row gt_center">10 (20%)</td>
<td class="gt_row gt_center">7 (13%)</td>
<td class="gt_row gt_center">0.4</td>
<td class="gt_row gt_center">0.8</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ATM</td>
<td class="gt_row gt_center">16 (15%)</td>
<td class="gt_row gt_center">11 (22%)</td>
<td class="gt_row gt_center">5 (9.3%)</td>
<td class="gt_row gt_center">0.079</td>
<td class="gt_row gt_center">0.4</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">EP300</td>
<td class="gt_row gt_center">15 (14%)</td>
<td class="gt_row gt_center">8 (16%)</td>
<td class="gt_row gt_center">7 (13%)</td>
<td class="gt_row gt_center">0.7</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">FBXW7</td>
<td class="gt_row gt_center">15 (14%)</td>
<td class="gt_row gt_center">11 (22%)</td>
<td class="gt_row gt_center">4 (7.4%)</td>
<td class="gt_row gt_center">0.038</td>
<td class="gt_row gt_center">0.4</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">PBRM1</td>
<td class="gt_row gt_center">15 (14%)</td>
<td class="gt_row gt_center">9 (18%)</td>
<td class="gt_row gt_center">6 (11%)</td>
<td class="gt_row gt_center">0.3</td>
<td class="gt_row gt_center">0.8</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">RBM10</td>
<td class="gt_row gt_center">15 (14%)</td>
<td class="gt_row gt_center">7 (14%)</td>
<td class="gt_row gt_center">8 (15%)</td>
<td class="gt_row gt_center">0.9</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">MGA</td>
<td class="gt_row gt_center">14 (14%)</td>
<td class="gt_row gt_center">10 (20%)</td>
<td class="gt_row gt_center">4 (7.5%)</td>
<td class="gt_row gt_center">0.065</td>
<td class="gt_row gt_center">0.4</td></tr>
    <tr><td class="gt_row gt_left" style="text-align: left; text-indent: 10px;">Unknown</td>
<td class="gt_row gt_center">2</td>
<td class="gt_row gt_center">1</td>
<td class="gt_row gt_center">1</td>
<td class="gt_row gt_center"></td>
<td class="gt_row gt_center"></td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CDKN2B.Del</td>
<td class="gt_row gt_center">14 (13%)</td>
<td class="gt_row gt_center">5 (9.8%)</td>
<td class="gt_row gt_center">9 (17%)</td>
<td class="gt_row gt_center">0.3</td>
<td class="gt_row gt_center">0.8</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">BAP1</td>
<td class="gt_row gt_center">13 (12%)</td>
<td class="gt_row gt_center">5 (9.8%)</td>
<td class="gt_row gt_center">8 (15%)</td>
<td class="gt_row gt_center">0.4</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">CDKN1A</td>
<td class="gt_row gt_center">12 (11%)</td>
<td class="gt_row gt_center">6 (12%)</td>
<td class="gt_row gt_center">6 (11%)</td>
<td class="gt_row gt_center">&gt;0.9</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">ERBB3</td>
<td class="gt_row gt_center">12 (11%)</td>
<td class="gt_row gt_center">6 (12%)</td>
<td class="gt_row gt_center">6 (11%)</td>
<td class="gt_row gt_center">&gt;0.9</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">KMT2C</td>
<td class="gt_row gt_center">12 (11%)</td>
<td class="gt_row gt_center">6 (12%)</td>
<td class="gt_row gt_center">6 (11%)</td>
<td class="gt_row gt_center">&gt;0.9</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">NF1</td>
<td class="gt_row gt_center">12 (11%)</td>
<td class="gt_row gt_center">6 (12%)</td>
<td class="gt_row gt_center">6 (11%)</td>
<td class="gt_row gt_center">&gt;0.9</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">BRCA2</td>
<td class="gt_row gt_center">11 (10%)</td>
<td class="gt_row gt_center">8 (16%)</td>
<td class="gt_row gt_center">3 (5.6%)</td>
<td class="gt_row gt_center">0.090</td>
<td class="gt_row gt_center">0.4</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">SMARCA4</td>
<td class="gt_row gt_center">11 (10%)</td>
<td class="gt_row gt_center">5 (9.8%)</td>
<td class="gt_row gt_center">6 (11%)</td>
<td class="gt_row gt_center">0.8</td>
<td class="gt_row gt_center">&gt;0.9</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">SPEN</td>
<td class="gt_row gt_center">11 (10%)</td>
<td class="gt_row gt_center">8 (16%)</td>
<td class="gt_row gt_center">3 (5.6%)</td>
<td class="gt_row gt_center">0.090</td>
<td class="gt_row gt_center">0.4</td></tr>
    <tr><td class="gt_row gt_left" style="font-weight: bold;">TSC1</td>
<td class="gt_row gt_center">11 (10%)</td>
<td class="gt_row gt_center">10 (20%)</td>
<td class="gt_row gt_center">1 (1.9%)</td>
<td class="gt_row gt_center">0.003</td>
<td class="gt_row gt_center">0.090</td></tr>
  </tbody>
  
  <tfoot class="gt_footnotes">
    <tr>
      <td class="gt_footnote" colspan="6"><sup class="gt_footnote_marks">1</sup> n (%)</td>
    </tr>
    <tr>
      <td class="gt_footnote" colspan="6"><sup class="gt_footnote_marks">2</sup> Pearson's Chi-squared test</td>
    </tr>
    <tr>
      <td class="gt_footnote" colspan="6"><sup class="gt_footnote_marks">3</sup> False discovery rate correction for multiple testing</td>
    </tr>
  </tfoot>
</table>
</div>
</div>
</div>
</section>
<section id="visualizing-data" class="level1">
<h1>Visualizing Data</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggvarclass</span>(mutations_clean) <span class="sc">+</span><span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="impact-example-script_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gggenecor</span>(mutations_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ `samples` argument is `NULL`. We will infer your cohort inclusion and resulting data frame will include all samples with at least one alteration in mutation, fusion or cna data frames</code></pre>
</div>
<div class="cell-output-display">
<p><img src="impact-example-script_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcomut</span>(mutations_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>ℹ `samples` argument is `NULL`. We will infer your cohort inclusion and resulting data frame will include all samples with at least one alteration in mutation, fusion or cna data frames</code></pre>
</div>
<div class="cell-output-display">
<p><img src="impact-example-script_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtopgenes</span>(mutations_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="impact-example-script_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="get-tmb" class="level2">
<h2 class="anchored" data-anchor-id="get-tmb">Get TMB</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>clin <span class="ot">&lt;-</span> <span class="fu">available_clinical_attributes</span>(<span class="st">"blca_nmibc_2017"</span>)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>clin<span class="sc">$</span>clinicalAttributeId <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "AGE"                     "CANCER_TYPE"            
 [3] "CANCER_TYPE_DETAILED"    "CONCOMITANT_CIS"        
 [5] "FRACTION_GENOME_ALTERED" "MUTATION_COUNT"         
 [7] "NUMBER_OF_TUMORS"        "ONCOTREE_CODE"          
 [9] "RECURRENCE"              "SAMPLE_COUNT"           
[11] "SEX"                     "SOMATIC_STATUS"         
[13] "SPECIMEN_TYPE"           "STAGE"                  
[15] "TMB_NONSYNONYMOUS"       "TREATMENT"              
[17] "TUMOR_SIZE"             </code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_clinical_by_study</span>(<span class="at">study_id =</span> <span class="st">"blca_nmibc_2017"</span>, <span class="st">"TMB_NONSYNONYMOUS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 105 × 8
   uniqueSampleKey         uniqu…¹ sampl…² patie…³ studyId clini…⁴ value dataL…⁵
   &lt;chr&gt;                   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;  
 1 UC0wMDAxNDUzLVQwMS1JTT… UC0wMD… P-0001… P-0001… blca_n… TMB_NO… 11.0… SAMPLE 
 2 UC0wMDAyMTY2LVQwMS1JTT… UC0wMD… P-0002… P-0002… blca_n… TMB_NO… 8.87… SAMPLE 
 3 UC0wMDAzMjM4LVQwMS1JTT… UC0wMD… P-0003… P-0003… blca_n… TMB_NO… 13.7… SAMPLE 
 4 UC0wMDAzMjU3LVQwMS1JTT… UC0wMD… P-0003… P-0003… blca_n… TMB_NO… 18.5… SAMPLE 
 5 UC0wMDAzMjYxLVQwMS1JTT… UC0wMD… P-0003… P-0003… blca_n… TMB_NO… 42.0… SAMPLE 
 6 UC0wMDAzMzUyLVQwMS1JTT… UC0wMD… P-0003… P-0003… blca_n… TMB_NO… 4.89… SAMPLE 
 7 UC0wMDAzMzYyLVQwMS1JTT… UC0wMD… P-0003… P-0003… blca_n… TMB_NO… 25.4… SAMPLE 
 8 UC0wMDAzMzcwLVQwMS1JTT… UC0wMD… P-0003… P-0003… blca_n… TMB_NO… 6.85… SAMPLE 
 9 UC0wMDAzMzg4LVQwMS1JTT… UC0wMD… P-0003… P-0003… blca_n… TMB_NO… 9.78… SAMPLE 
10 UC0wMDAzMzkxLVQwMS1JTT… UC0wMD… P-0003… P-0003… blca_n… TMB_NO… 8.80… SAMPLE 
# … with 95 more rows, and abbreviated variable names ¹​uniquePatientKey,
#   ²​sampleId, ³​patientId, ⁴​clinicalAttributeId, ⁵​dataLevel</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>